### PR TITLE
Spider "white spindle"

### PR DESCRIPTION
**Pull request by Arteben**
## About the PR

Adding a tamed spider that spawns webs using its hunger. Also here adding a cargo crate with it.

## Why / Balance

This is a way for primitives to obtain fabric from spider silk.
https://docs.google.com/document/d/1dxHC4NndMCGsJyS7QL7E_uk3489LdXdSPIi2OnLXd0U/edit?usp=sharing

## Technical details

I changed component EggLayerComponent added new flag "manySpawnsForbidden" and a bit fixed/refactored it.
## Media

image

[x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

EggLayerComponent
EggLayerSystem
and tranlates for it

**Changelog**

:cl:
- add: Added a hand spider "White spindle"!

### CODE REVIEW

File: Content.Server/Animals/Components/EggLayerComponent.cs [hunk 1]
```diff
@@ -12,6 +12,9 @@ namespace Content.Server.Animals.Components;
 [RegisterComponent]
 public sealed partial class EggLayerComponent : Component
 {
+    /// <summary>
+    ///     This is action that available for player in actions list
+    /// </summary>
     [DataField("eggLayAction", customTypeSerializer: typeof(PrototypeIdSerializer<EntityPrototype>))]
```
Comment (TheShuEd) [inline Content.Server/Animals/Components/EggLayerComponent.cs:18]: naming in brekets is obsolete. Use `[DataField, custom...`

File: Content.Server/Animals/Components/EggLayerComponent.cs [hunk 2]
```diff
@@ -49,5 +52,12 @@ public sealed partial class EggLayerComponent : Component
     [DataField("accumulatedFrametime")]
```
Comment (TheShuEd) [inline Content.Server/Animals/Components/EggLayerComponent.cs:52]: same

File: Content.Server/Animals/Components/EggLayerComponent.cs [hunk 3]
```diff
@@ -49,5 +52,12 @@ public sealed partial class EggLayerComponent : Component
...
     public float AccumulatedFrametime;
 
+    /// <summary>
+    ///     Prohibits creating more than one element in one place
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("manySpawnsForbidden")]
```
Comment (TheShuEd) [inline Content.Server/Animals/Components/EggLayerComponent.cs:59]: same

File: Content.Server/NPC/Queries/Queries/ComponentWithTagsQuery.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Content.Server.NPC.Queries.Queries;
+
+/// <summary>
+/// Returns nearby components that match the specified components
+/// and match all tags.
+/// </summary>
+public sealed partial class ComponentWithTagsQuery : UtilityQuery
+{
+    [DataField("components", required: true)]
```
Comment (TheShuEd) [inline Content.Server/NPC/Queries/Queries/ComponentWithTagsQuery.cs:11]: same

File: Content.Server/NPC/Queries/Queries/ComponentWithTagsQuery.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    public ComponentRegistry Components = default!;
+
+    [DataField("tags", required: true)]
```
Comment (TheShuEd) [inline Content.Server/NPC/Queries/Queries/ComponentWithTagsQuery.cs:14]: same

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml [hunk 6]
```diff
@@ -207,3 +207,13 @@
...
   category: Livestock
   group: market
+
+- type: cargoProduct
+  id: LivestockWhiteSpindle
+  icon:
+    sprite: Mobs/Animals/white_spider.rsi
+    state: white
+  product: CrateNPCWhiteSpindle
+  cost: 3900
```
Comment (TheShuEd) [inline Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:217]: I'm not sure, but I think it's very expensive to produce silk, which is not very sought after yet.

File: Resources/Textures/Mobs/Animals/white_spider.rsi/meta.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+          1,
+          1
+        ]
+      ]
+    },
+    {
+      "name": "white_dead",
+      "delays": [
+        [
+          1
```
Comment (TheShuEd) [inline Resources/Textures/Mobs/Animals/white_spider.rsi/meta.json:40]: unnecessarily

**Review by TheShuEd — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml (hunk 6 above)
Comment (Arteben) [reply Resources/Prototypes/Catalog/Cargo/cargo_livestock.yml:217]: Ii is a bit expensive than a cow, but it is an exotic animal. Food is more necessary and these spiders can spawn webs anywhere and annoying everyone so.. for my opinion the price is normal.

**Pull request closed without merging**